packages/components/src/components/post-header/post-header.tsx

```diff
@@ -147,12 +147,35 @@ export class PostHeader {
       0,
       this.scrollParent instanceof Document
         ? this.scrollParent.documentElement.scrollTop
-        : this.scrollParent.scrollTop,
+        : this.scrollParent.scrollTop
     );
-
+  
     this.host.style.setProperty('--header-scroll-top', `${st}px`);
-  }
-
+  
+    if (this.device !== 'desktop') {
+      this.host.style.setProperty('--total-header-height', `${this.host.getBoundingClientRect().height}px`);
+      return;
+    }
+  
+    const computedStyles = getComputedStyle(this.host);
+  
+    const getCSSValue = (property: string, fallback: number) =>
+      parseFloat(computedStyles.getPropertyValue(property)) || fallback;
+  
+    const globalHeaderHeight = getCSSValue('--global-header-height', 72);
+    const globalHeaderReducedHeight = getCSSValue('--global-header-reduced-height', 24);
+    const localHeaderHeight = getCSSValue('--local-header-height', 112);
+    const localHeaderMinHeight = getCSSValue('--local-header-min-height', 56);
+  
+    const maxGlobalScroll = globalHeaderHeight - globalHeaderReducedHeight;
+    const scrolledPastGlobal = Math.max(0, st - maxGlobalScroll);
+  
+    const newGlobalHeaderHeight = Math.max(globalHeaderHeight - Math.min(st, maxGlobalScroll), globalHeaderReducedHeight);
+    const newLocalHeaderHeight = Math.max(localHeaderHeight - scrolledPastGlobal, localHeaderMinHeight);
+  
+    this.host.style.setProperty('--total-header-height', `${newGlobalHeaderHeight + newLocalHeaderHeight}px`);
```

Comment:
There is already a CSS variable for that: https://github.com/swisspost/design-system/blob/fd38fbfd01c6cd02d81a52580f8e2dcd696578cb/packages/styles/src/components/header/index.scss#L8

Just make sure it is up-to-date for all header configuration (full height/reduced, desktop/tablet/mobile, with/without local header, long wrapping application title) but then you can directly use it.